### ISSUE TITLE
Refactoring: Analyzing Abstract_Check_Runner Class for Complexity Reduction

### ISSUE THREAD

**Issue by mukeshpanchal27** [10up/plugin-check#229]
### Is your enhancement related to a problem? Please describe.

Following up on PR #228, during the code review process, we encountered a warning related to excessive class complexity in the file `plugin-check/includes/Checker/Abstract_Check_Runner.php` at line 19. The warning message received was:

```
ExcessiveClassComplexity: The class Abstract_Check_Runner has an overall complexity of 54 which is very high. The configured complexity threshold is 50.
```

To address this warning, it is recommended that we refactor the code of the `Abstract_Check_Runner` class to reduce its complexity. Currently, we have temporarily suppressed the warning by adding `@SuppressWarnings(PHPMD.ExcessiveClassComplexity)` in the class docblock.

This issue aims to investigate and analyze the `Abstract_Check_Runner` class in order to identify areas where complexity can be reduced.

Please feel free to provide any additional insights or suggestions for the refactoring process.

cc. @swissspidy 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct
